stacks-signer/src/v0/signer.rs

```diff
@@ -513,8 +507,39 @@ impl Signer {
             BlockResponse::Rejected(block_rejection) => {
                 self.handle_block_rejection(block_rejection);
             }
+        };
+
+        // Remove this block validation from the pending table
+        let signer_sig_hash = block_response.signer_signature_hash();
+        self.signer_db
+            .remove_pending_block_validation(&signer_sig_hash)
```

Comment:
I think you could maybe have mark_locally or mark_globally etc. functions to just auto update these pending lists because if we have come to a consensus, it doens't matter what the block response is. Does that make sense?

Reply:
Interesting, good call. That somewhat contrasts the ticket for [always sending BlockResponses](https://github.com/stacks-network/stacks-core/issues/5451), but if the block already reached consensus, there's no point in adding load to our node with a new block response. Your suggestion would probably also help mitigate the issue that this PR tries to solve.

Reply:
I think neither of us noticed this was in `handle_block_response`, not `handle_block_validation_response` 🤦. I've moved it to `handle_block_validation_response`.

Regarding your point, though, I think we should remove the pending validation when a block is marked globally accepted/rejected. If the validation response comes in after it's marked as globally handled (the "typical" scenario), we'll still respond with a block response, which we want. But if our signer is "catching up", and the pending block validation is 1 or more blocks behind the tip, we should just drop it.